scripts/dropbox/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,262 @@
+            ))
+        else:
+            if new:
+                external_accounts_created += 1
+            linked_node_settings_documents = database['dropboxnodesettings'].find({
+                'user_settings': user_settings_document['_id']
+            })
+            if not user or not user.is_active:
+                if linked_node_settings_documents.count() and not user.is_merged:
+                    logger.warn("DropboxUserSettings<_id:{0}> has no owner, but is used by DropboxNodeSettings: {1}.".format(
+                        user_settings_document['_id'],
+                        ', '.join([each['_id'] for each in linked_node_settings_documents])
+                    ))
+                    raise RuntimeError("This should never happen.")
+                else:
+                    logger.info("DropboxUserSettings<_id:{0}> either has no owner or the owner's account is not active, and will not be migrated.".format(
+                        user_settings_document['_id']
+                    ))
+                    continue
+            else:
+                user_settings_instance = make_new_user_settings(user)
+                for node_settings_document in linked_node_settings_documents:
+                    verify_node_settings_document(node_settings_document)
+                    if node_settings_document['deleted']:
+                        logger.info(
+                            "Found dropboxnodesettings document (id:{0}) that is marked as deleted.{1}".format(
+                                node_settings_document['_id'],
+                                rm_msg
+                            )
+                        )
+                        continue
+                    node = Node.load(node_settings_document['owner'])
+                    if not node:
+                        logger.info("DropboxNodeSettings<_id:{0}> has no associated Node, and will not be migrated.".format(
+                            node_settings_document['_id']
+                        ))
+                        continue
+                    else:
+                        make_new_node_settings(
+                            node,
+                            node_settings_document,
+                            external_account,
+                            user_settings_instance
+                        )
+                        migrated_node_settings += 1
+        migrated_user_settings += 1
+    logger.info(
+        "Created {0} new external accounts, migrated {1} dropboxusersettings, and migrated {2} dropboxnodesettings.".format(
+            external_accounts_created, migrated_user_settings, migrated_node_settings
+        )
+    )
+    if remove_old:
+        remove_old_documents(
+            old_user_settings, old_user_settings_count,
+            old_node_settings, old_node_settings_count,
+            dry_run
+        )
+        
+    if dry_run:
+        raise RuntimeError('Dry run, transaction rolled back.')
+
+def main():
```

Comment:
A file handler should be added for the logger, so that logs get written to disk. See https://cosdev.readthedocs.org/en/latest/osf/migrations.html